GANDLF/utils/tensor.py

```diff
@@ -220,7 +220,45 @@ def send_model_to_device(model, amp, device, optimizer):
     return model, amp, device
 
 
-def get_class_imbalance_weights(training_data_loader, parameters):
+def get_class_imbalance_weights_classification(training_df, params):
+    """
+    This function calculates the penalty used for loss functions in multi-class problems.
+    It looks at the column "valuesToPredict" and identifies unique classes, fetches the class distribution
+    and generates the required class weights, and then generates the weights needed for loss function which
+    are inverse of the class weights generated divided by the total number of items in a single column
+
+    Args:
+        training_Df (pd.DataFrame): The training data frame.
+        parameters (dict) : The parameters passed by the user yaml.
+
+    Returns:
+        dict: The penalty weights for different classes under consideration for classification.
+
+    """
+    class_count = training_df["ValueToPredict"].value_counts().to_dict()
+    total_count = len(training_df)
+
+    penalty_dict, weight_dict = {}, {}
+    for i in range(params["model"]["num_classes"]):
+        penalty_dict[i], weight_dict[i] = 0, 0
+
+    for label in class_count.keys():
+        weight_dict[label] = class_count[label] / total_count
+
+    for label in class_count.keys():
+        penalty_dict[label] = total_count / class_count[label]
+
+    penalty_sum = np.fromiter(penalty_dict.values(), dtype=np.float64).sum()
+
+    print("Normalizing penalty dict for each class")
+
+    for label in class_count.keys():
+        penalty_dict[label] = penalty_dict[label] / penalty_sum
+
+    return weight_dict, penalty_dict
```

Comment:
This is the usage of the function:

```python
        if params["problem_type"] == "classification":
            (
                params["weights"],
                params["class_weights"],
            ) = get_class_imbalance_weights_classification(training_data, params)
```
But the return is 
```python
return weight_dict, penalty_dict
```
I think either the return or the usage should be reversed. Right?

Reply:
I see that the [CEL loss function](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/losses/regression.py#L6) is using `"class_weights"`, but the [MCD loss](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/losses/segmentation.py#L83) is using `"weights"`. We should change the CEL function and change the return of `get_class_imbalance_weights_classification` to `return penalty_dict, weight_dict`. 

What do you think?